platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCompactor.java

```diff
@@ -438,6 +442,18 @@ public boolean compact() throws IOException, InterruptedException {
                     (filesToCompactSize - compactedFilesSize) * UnitConstants.BYTES_TO_MEBIBYTES);
         }
 
+        if (reportTotalSpaceMetricFunction != null) {
```

Comment:
Is this the only place where to update "total space" metrics? What about MerkleDbDataSource.saveRecords(), which creates files on compaction level 0?

Reply:
The question of how often we want these metrics to be reported. Right now it's implemented so that it's reported every compaction. In this case we'll get the best case scenario of the space usage because most of the default compaction level files should be eliminated. If we add it to saveRecords(), we'll see some dynamics of the default compaction level.

Reply:
It makes sense, I'll add it.

Reply:
I ended up with adding it into `DataFileCompactor#compact` at the very beginning. Here is the logic:
1. We want to report this metric every time we save records to the disk
2. Method `DataFileCompactor#compact` is invoked every time we save records to the disk but we don't always run the compaction because there may be not enough files to compact. Therefore, it a good place to access the size of the files prior to compaction.